packages/main/src/Menu.ts

```diff
@@ -266,22 +273,73 @@ class Menu extends UI5Element {
 		return this.shadowRoot!.querySelector<ResponsivePopover>("[ui5-responsive-popover]")!;
 	}
 
+	get _list() {
+		return this.shadowRoot!.querySelector<List>("[ui5-list]")!;
+	}
+
+	get _menuItemGroups() {
+		return this.items.filter((item) : item is MenuItemGroup => item.isGroup);
+	}
+
+	// Return only the menu items, excluding separators and items that belong to groups
 	get _menuItems() {
-		return this.items.filter((item): item is MenuItem => !item.isSeparator);
+		return this.items.filter((item) : item is MenuItem => !item.isSeparator && !item.isGroup);
+	}
+
+	get _menuItemsAll() {
+		const items: MenuItem[] = [];
+
+		this.items.forEach(item => {
+			if (item.isGroup) {
+				items.push(...(item as MenuItemGroup)._menuItems);
+			} else if (!item.isSeparator) {
+				items.push(item as MenuItem);
+			}
+		});
+
+		return items;
+	}
+
+	get _menuItemsNavigation() {
+		const parent = this._parent || this;
```

Comment:
Can we think of a way without using ref to the parent?

Reply:
As I already mentioned - this getter returns `items` (children) of the specific component, which is their ...umm... parent.

Reply:
As above - I will remove it.